builtin/providers/openstack/resource_openstack_fw_firewall_v1.go

```diff
@@ -57,6 +58,18 @@ func resourceFWFirewallV1() *schema.Resource {
 				ForceNew: true,
 				Computed: true,
 			},
+			"associated_routers": &schema.Schema{
+				Type:          schema.TypeSet,
```

Comment:
Does the API return the associated routers as an array in the same order as what the user gave? If so, this can be a `TypeList`. But if the API takes it upon itself to re-order (for example, alphabetical), then `TypeSet` is the way to go.